payloads/v9/channel.ts

```diff
@@ -1535,6 +1571,10 @@ export interface APIBaseComponent<T extends ComponentType> {
 	 * The type of the component
 	 */
 	type: T;
+	/**
+	 * int32, auto generated via increment if not provided
+	 */
+	id?: number;
```

Comment:
Not sure how discord-api-types would want to model it, but `id` is optional on request, but will _always_ be provided as part of the response.

Reply:
We have other places where that's an issue, usually we either create rest-specific types from the api ones or document it as always present when received